### DISCUSSION TITLE
課題14について（9/5）

### DISCUSSION THREAD

**Discussion by TakedaAyano** [nsc-kiwamu/traning-takeda#73] [Q&A]
①問題１が正しくコンソールに出力されましたので、
問題２を問題１と同じようにfor文で式を作ってみたのですが、コンソールに出力されません。
この原因はなぜでしょうか？同じようにしているのに出ないため原因がわかりません。

②ちなみになのですが、judgeSosuuを作ってみたのですが、こちらは合っていますでしょうか？
/**
  * 素数の判断をする
  * 素数の場合は引数で受け取った値を返却
  * 素数でない場合はundedefindを返却
  * @param target 素数判定したい値
  * @return 素数判定結果
  */
 function judgeSosuu(target: number):number | undefined {
  let retNum:number | undefined

  // 素数判定の処理を追加する
  for(let i=2; i < target; i++){ // 2から渡された数まで繰り返す。
        if(target % i == 0){ // target(受け取った値）÷iの余りがなければ、
          return(retNum)   
    }
  }
  return target
}

**Comment by nsc-kiwamu** [comment]
回答1
「問題２を問題１と同じようにfor文で式を作ってみたのですが、コンソールに出力されません。」
→「コンソールには出力されるが、想定通りの値が出ない」で合っていますか？
　（その前提で回答しますが）

「解析のためにコンソールログを出して確認する」
はできていますか？
for文の中で変数iの値をコンソール出力すれば、ソースが誤っていることがわかると思います。

回答2
judgeSosuuを作ってみたのですが、こちらは合っていますでしょうか？
例えば、問題1では「1～100」の数値のうち素数をコンソール出力する。
という内容で、これがいわゆる仕様です。

合っているかどうか(=仕様を満たしているかどうか)は確認して欲しいです。
具体的には、コンソール出力された値が以下のサイト(他のサイトでもなんでも良いです)
の結果と一致するかなどを見る。
https://terakoya.ameba.jp/a000001446/

**Reply by TakedaAyano** [reply]
ご確認ありがとうございます。

回答１について
・すみません、詳しく内容をかけていませんでした。
問題2を実行すると、以下のエラーメッセージが表示されてしまいます。
課題13から課題14へ正しくインポートされていないからでしょうか？

$ node dist/traning14/Practice14.js
node:internal/errors:465
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'C:\worktraning\traning-takeda\dist\traning13\Common13' imported from C:\worktraning\traning-takeda\dist\traning14\Practice14.
js
?[90m    at new NodeError (node:internal/errors:372:5)?[39m
?[90m    at finalizeResolution (node:internal/modules/esm/resolve:437:11)?[39m
?[90m    at moduleResolve (node:internal/modules/esm/resolve:1009:10)?[39m
?[90m    at defaultResolve (node:internal/modules/esm/resolve:1218:11)?[39m
?[90m    at ESMLoader.resolve (node:internal/modules/esm/loader:580:30)?[39m
?[90m    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:294:18)?[39m
?[90m    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:80:40)?[39m
?[90m    at link (node:internal/modules/esm/module_job:78:36)?[39m {
  code: ?[32m'ERR_MODULE_NOT_FOUND'?[39m
}


回答２について
・承知いたしました。
確認したところ、問題１は１～１００までの素数だけがコンソールに出力されておりました。
